source/feature/openvswitch/openvswitch-spec.html.md

```diff
@@ -0,0 +1,183 @@
+The OVS rollback is to be integrated into the existing rollback functionality without heavely affecting existing beahaviours:
+
+- OVS configurators should trigger a rollback which is to be handled at the api level (as described above).
+- The ifcfg configurator should keep the existing logic, but in addition it should inform the api level of the rollback which it handled.
+Moving the handling of the rollback from the ifcfg configurator to the api level should be investigated, simplifying rollback logic.
+- The api level should combine both ifcfg and ovs results in terms of the rollback, and issue relevant commands. TBD
+
+### VDSM API
+Minimal changes in current API verbs are required:
+
+- setupNetworks: Per network, add a net-switch type.
+  - Currently it can have two values: ‘legacy’ and ‘ovs’.
+  - If not explicitly set, assume ‘legacy’. (for compatability with Endgine 3.6)
+- oVirt currently differentiates between VM (bridged) networks and non-VM networks due to historical performance benefit of non-bridged networks for non-VM payloads (such as storage and migration). This differentiation is moot with OVS, but in order to keep Engine oblivious to the change, Vdsm should lie to it, and report bridged=False for networks that requested it. Dropping the differentiation from Engine should happen in a later stage.
+
+### Upgrade from native to OVS networking
+Transition between networking implementation must be supported for the host as a whole. Nevertheless, if the user has defined his own networking configuration which is not controlled by VDSM, it should coexist with VDSM controlled networks.
+
+Mixing between native and OVS controlled networks is not initially supported, therefore, when a network is marked with a different type from the existing, a validation check should make sure that all other networks are marked for the same network type.
+
+The following sequence describes the transition steps @Engine:
+
+- Hosts are upgraded to OVS supported version (4.0).
+- When all active hosts in the cluster have been upgraded, mark the cluster as ‘4.0’ where OVS is supported.
+- The default network type for the cluster is set to OVS (for new hosts).
+- Update individual hosts to transition to OVS: 
+(TBD: one or all of the options below should be chosen)
+  - The host networking can be changed when no VM/s are connected to it. With this option oVirt must support VM migration between different switch types.
+  - The host networking can be changed in-place without the need to migrate the VM/s. With this option, migration between hosts that have different networking types is not needed (but can coexist).
+
+The following sequence describes the transition steps @Host:
+- Receive a setupNetworks verb.
+- Detect that a network has been marked with a type that is different from the current.
+- Validate that all existing networks exist in the command and that all are marked with the same type.
+- Tear down the networks.
```

Comment:
Regarding the bonding comment: At the moment, we are not supposed to have a net and a bond in different switches, so we can use the net switch type (as they are linked by usage).
Is there a scenario where this may not be true?